airbyte-integrations/connectors/destination-gcs/src/test/java/io/airbyte/integrations/destination/gcs/avro/GcsAvroFormatConfigTest.java

```diff
@@ -0,0 +1,167 @@
+  }
+
+  @Test
+  public void testParseCodecConfigXz() {
+    // default compression level 6
+    CodecFactory codecFactory1 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize("{ \"codec\": \"xz\" }"));
+    assertEquals("xz-6", codecFactory1.toString());
+
+    // compression level 7
+    CodecFactory codecFactory2 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize("{ \"codec\": \"xz\", \"compression_level\": 7 }"));
+    assertEquals("xz-7", codecFactory2.toString());
+  }
+
+  @Test
+  public void testParseCodecConfigZstandard() {
+    // default compression level 3
+    CodecFactory codecFactory1 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize("{ \"codec\": \"zstandard\" }"));
+    // There is no way to verify the checksum; all relevant methods are private or protected...
+    assertEquals("zstandard[3]", codecFactory1.toString());
+
+    // compression level 20
+    CodecFactory codecFactory2 = S3AvroFormatConfig.parseCodecConfig(
+        Jsons.deserialize(
+            "{ \"codec\": \"zstandard\", \"compression_level\": 20, \"include_checksum\": true }"));
+    // There is no way to verify the checksum; all relevant methods are private or protected...
+    assertEquals("zstandard[20]", codecFactory2.toString());
+  }
+
+  @Test
+  public void testParseCodecConfigSnappy() {
+    JsonNode snappyConfig = Jsons.deserialize("{ \"codec\": \"snappy\" }");
+    CodecFactory codecFactory = S3AvroFormatConfig.parseCodecConfig(snappyConfig);
+    assertEquals(DataFileConstants.SNAPPY_CODEC, codecFactory.toString());
+  }
+
+  @Test
+  public void testParseCodecConfigInvalid() {
+    try {
+      JsonNode invalidConfig = Jsons.deserialize("{ \"codec\": \"bi-directional-bfs\" }");
+      S3AvroFormatConfig.parseCodecConfig(invalidConfig);
+      fail();
+    } catch (IllegalArgumentException e) {
+      // expected
+    }
+  }
+
+  @Test
+  public void testHandlePartSizeConfig() throws IllegalAccessException {
+
+    JsonNode config = ConfigTestUtils.getBaseConfig(Jsons.deserialize("{\n"
+        + "  \"format_type\": \"AVRO\",\n"
+        + "  \"part_size_mb\": 6\n"
+        + "}"));
+
+    GcsDestinationConfig gcsDestinationConfig = GcsDestinationConfig
+        .getGcsDestinationConfig(config);
+    ConfigTestUtils.assertBaseConfig(gcsDestinationConfig);
+
+    S3FormatConfig formatConfig = gcsDestinationConfig.getFormatConfig();
+    assertEquals("AVRO", formatConfig.getFormat().name());
+    assertEquals(6, formatConfig.getPartSize());
+    // Assert that is set properly in config
+    StreamTransferManager streamTransferManager = S3StreamTransferManagerHelper.getDefault(
+        gcsDestinationConfig.getBucketName(), "objectKey", null,
+        gcsDestinationConfig.getFormatConfig().getPartSize());
+
+    Integer partSizeBytes = (Integer) FieldUtils.readField(streamTransferManager, "partSize", true);
+    assertEquals(6291456, partSizeBytes);
```

Comment:
could you possibly create a constant or local variable that explains this "magic number"?

Reply:
updated, thanks